File: src/ClientApp/models/letsEncryptServer/account/requests/PatchAccountRequest.ts
import { PatchAction } from '@/models/PatchAction'

export interface PatchHostnameRequest {
  hostname?: PatchAction<string>
  isDisabled?: PatchAction<boolean>
}

export interface PatchAccountRequest {
  description?: PatchAction<string>
  isDisabled?: PatchAction<boolean>
  contacts?: PatchAction<string>[]
  hostnames?: PatchHostnameRequest[]
}

File: src/MaksIT.CertsUI.Client/Models/PatchAccountRequest.cs
namespace MaksIT.CertsUI.Client.Models;

public class PatchAccountRequest {
  /// <summary>Per-field patch operations (keys are camelCase to match the API).</summary>
  public Dictionary<string, int>? Operations { get; set; }

  public string? Description { get; set; }
  public bool? IsDisabled { get; set; }
  public List<string>? Contacts { get; set; }
  public List<PatchHostnameRequest>? Hostnames { get; set; }
}
